app/src/main/java/Bamboo/view/game/Canvas.java

```diff
@@ -96,12 +105,14 @@ private void colorTile(Tile tile, Color color, Graphics2D g2d){
         v = VectorConverter.doubleAndOffsetOddRows(v);
         int x = centreX + (v.getQ() * circle_radius/2) ;
         int y = centreY + (v.getR() * circle_radius/2) ;
+        g2d.drawImage(images.get(color),x,y,circle_radius,circle_radius,null);
+
 
-        g2d.setStroke(tile.getCircle_thickness());
-        g2d.setColor(color);
-        g2d.fillOval(x,y,circle_radius,circle_radius);
-        g2d.setColor(tile.getOutline());
-        g2d.drawOval(x, y, circle_radius, circle_radius);
+        //g2d.setStroke(tile.getCircle_thickness());
+        //g2d.setColor(color);
+        //g2d.fillOval(x,y,circle_radius,circle_radius);
+        //g2d.setColor(tile.getOutline());
+        //g2d.drawOval(x, y, circle_radius, circle_radius);
```

Comment:
If this is being left in in case the image is not found (which is a great idea) then you need to write the logic where it deals with the error using a try/catch statement:
```
try{
  drawImage()
}
catch(IOException e)
{
    drawCircle()
}
```